components/upload/utils.tsx

```diff
@@ -110,6 +110,15 @@ export function previewImage(file: File | Blob): Promise<string> {
 
       resolve(dataURL);
     };
-    img.src = window.URL.createObjectURL(file);
+    img.crossOrigin = "anonymous";
+    if (file.type.startsWith("image/svg+xml")) {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        if (reader.result) img.src = reader.result as string;
+      });
+      reader.readAsDataURL(file);
```

Comment:
`reader.readAsDataURL(file)` seems can be used as a general replacement of `window.URL.createObjectURL`. Some doubt the performance but it's debatable. Also currently `window.URL.revokeObjectURL` isn't being used.

Should it be used for all image types directly (drop the if-else)?

Reply:
Right, we should use readAsDataURL and drop createObjectURL.

Reply:
We replace readAsDataURL to createObjectURL for perfermance issue in https://github.com/ant-design/ant-design/pull/15984

https://stackoverflow.com/a/31743665/3040605

Seems that we should call `revokeObjectURL` after remove file item.